kratos/input_output/vtk_output.cpp

```diff
@@ -120,104 +127,71 @@ void VtkOutput::WriteConditionsAndElements(const ModelPart &rModelPart, std::ofs
     {
-        const auto& condition_geometry = condition.GetGeometry();
-        const unsigned int number_of_nodes = condition_geometry.size();
+        const auto& entity_geometry = entity->GetGeometry();
+        const unsigned int number_of_nodes = entity_geometry.size();
 
         WriteScalarDataToFile((unsigned int)number_of_nodes, rFileStream);
-        for (const auto& node : condition_geometry){
+        for (const auto& node : entity_geometry){
             if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << " ";
             int id = mKratosIdToVtkId[node.Id()];
             WriteScalarDataToFile((int)id, rFileStream);
         }
         if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
     }
-
 }
 
 void VtkOutput::WriteConditionAndElementTypes(const ModelPart &rModelPart, std::ofstream& rFileStream)
 {
+
     const auto& local_mesh = rModelPart.GetCommunicator().LocalMesh();
     // write cell types header
     rFileStream << "\nCELL_TYPES " << local_mesh.NumberOfConditions() + local_mesh.NumberOfElements() << "\n";
+    WriteCellType<Element> (local_mesh.Elements().GetContainer(), rFileStream);
+    WriteCellType<Condition> (local_mesh.Conditions().GetContainer(), rFileStream);
+}
 
-    // write elements types
-    for (const auto& elem : local_mesh.Elements())
-    {
-        int  cell_type = -1;
-
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle2D3)
-             cell_type = 5; // triangle 2d
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4)
-             cell_type = 9; // quad 2d
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4)
-             cell_type = 10; // tetra 3d
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2)
-             cell_type = 3; // line 2d or 3d
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line3D2)
-             cell_type = 3; // line 2d or 3d
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point2D)
-             cell_type = 1;
-        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point3D)
-             cell_type = 1;
-        if( cell_type < 0)
-            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
-
-        WriteScalarDataToFile( (int)cell_type, rFileStream);
-        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
-    }
-
-    // write conditions types
-    for (const auto& condition : local_mesh.Conditions())
+template <typename TEntity>
+void VtkOutput::WriteCellType(const typename PointerVectorSet<TEntity, IndexedObject>::ContainerType& rContainer, std::ofstream& rFileStream)
```

Comment:
I mean like this:
https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MappingApplication/custom_utilities/mapper_utilities.h#L226-L241
Then you don't have to deal with the details of how the Container is implemented

Reply:
Didnt know that is possible. I am used to this way of doing it. But this looks much simple.